src/SWP/TemplateEngineBundle/Controller/ContainerController.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Template Engine Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\TemplateEngineBundle\Controller;
+
+use FOS\RestBundle\Controller\FOSRestController;
+use Nelmio\ApiDocBundle\Annotation\ApiDoc;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use FOS\RestBundle\Controller\Annotations as Rest;
+use SWP\ContentBundle\Factory\KnpPaginatorRepresentationFactory;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\UnprocessableEntityHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use SWP\TemplateEngineBundle\Form\Type\ContainerType;
+
+class ContainerController extends FOSRestController
+{
+    /**
+     * Lists all registered containers
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Lists all registered containers",
+     *     statusCodes={
+     *         200="Returned on success."
+     *     }
+     * )
+     * @Route("/api/{version}/templates/containers/", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_templates_list")
+     * @Method("GET")
+     * @Rest\View(statusCode=200)
+     */
+    public function listAction(Request $request)
+    {
+        $entityManager = $this->get('doctrine')->getManager();
+        $paginator = $this->get('knp_paginator');
+        $containers = $paginator->paginate($entityManager->getRepository('SWP\TemplateEngineBundle\Model\Container')->getAll());
```

Comment:
https://github.com/m038/web-renderer/blob/SWP-69/src/SWP/ContentBundle/Storage/DoctrineStorage.php
https://github.com/m038/web-renderer/blob/SWP-69/src/SWP/ContentBundle/Storage/Doctrine/PHPCRStorage.php

@ahilles107  I think this is what @takeit is talking about.